### ISSUE TITLE
UnicodeDecodeError with read_sav, cannot specify locale / encoding

### ISSUE THREAD

**Issue by JessicaKlos** [stspec/pyspssio#6]
Hi,
I try to read an sav dataset which is not in standard UTF-8 encoding with pyspssio.read_sav.
With the default settings, I get a unicode decode error:
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa4 in position 3: invalid start byte

With the pyreadstat.read_sav function specifying encoding='iso-8859-1' would solve the issue.
Is there a similar option for the pyspssio.read_sav function? I tried specifying different settings for the locale parameter (e.g. 'iso-8859-1', 'en', 'de'), but then I always get this error: Error: unsupported locale setting

Thank you for your support!

**Comment by stspec** [comment]
Try setting the `unicode=False` argument in addition to supplying your locale. Also, I believe iso-8859-1 is just an encoding and not a specific locale (several locales use it). Since pyreadstat is decoding the binary data directly, it can use the encoding without worrying about a specific locale. The spss i/o module requires a specific locale. 

My recommendation would be to first make sure you're able to set the locale using Python's built-in [locale](https://docs.python.org/3/library/locale.html) module.

Also, this thread might be useful. Try "de_DE"? https://stackoverflow.com/questions/3191664/list-of-all-locales-and-their-short-codes#3191729

**Comment by JessicaKlos** [comment]
Hi,
Thank you for you quick answer!
Sorry, where do I set the unicode=False? The read_sav function does not know this parameter:
TypeError: read_sav() got an unexpected keyword argument 'unicode'
I also tried locale='de_DE' or 'de_de' and other options from the thread you shared and from the locale list returned by locale.locale_alias. But everything returns the same error (Error: unsupported locale setting).

**Comment by stspec** [comment]
Try setting the `unicode=False` argument in addition to supplying your locale. Also, I believe iso-8859-1 is just an encoding and not a specific locale (several locales use it). Since pyreadstat is decoding the binary data directly, it can use the encoding without worrying about a specific locale. The spss i/o module requires a specific locale. 

My recommendation would be to first make sure you're able to set the locale using Python's built-in [https://docs.python.org/3/library/locale.html](locale) module.

Also, this thread might be useful. Try "de_DE"? https://stackoverflow.com/questions/3191664/list-of-all-locales-and-their-short-codes#3191729

**Comment by stspec** [comment]
> Hi,
> Thank you for you quick answer!
> Sorry, where do I set the unicode=False? The read_sav function does not know this parameter:
> TypeError: read_sav() got an unexpected keyword argument 'unicode'
> I also tried locale='de_DE' or 'de_de' and other options from the thread you shared and from the locale list returned by locale.locale_alias. But everything returns the same error (Error: unsupported locale setting).

Unfortunately, I'm traveling right now so can't really investigate further at the moment. Apparently locale_alias might not be entirely accurate though and its return values can't be used directly in setlocale.

I would suggest trying to use the `Reader` class directly as a workaround for now. Basically, you can try to operate in Unicode (don't supply a locale), then set the encoding directly. For example...

```
with pyspssio.Reader(filename) as sav:
    sav.encoding = "iso-8859-1"
    metadata = sav.metadata
    df = sav.read_data() 
```

**Comment by stspec** [comment]
@JessicaKlos - I just released v0.4.3 to improve locale handling. If you're on Windows, I think you can use `deu_deu`, `German`, or `German_Germany.1252` as valid German locales.

Closing this issue for now. Feel free to open a feature request if you think this package should allow specifying the encoding directly.
